evap/rewards/tools.py

```diff
@@ -71,14 +87,20 @@ def grant_reward_points(user, semester):
     required_courses = Course.objects.filter(participants=user, semester=semester, is_required_for_reward=True)
     if not required_courses.exists():
         return False
-    # does the user not participate in any more required courses in this semester?
-    if required_courses.exclude(voters=user).exists():
-        return False
-    # did the user not already get reward points for this semester?
-    if RewardPointGranting.objects.filter(user_profile=user, semester=semester).exists():
+
+    voted_courses_count = required_courses.filter(voters=user).count()
+    # full evaluation progress from 0.0 to 1.0
+    progress = float(voted_courses_count) / float(required_courses.count())
+
+    # How many points have been granted to this user this semester?
+    granted_points = RewardPointGranting.objects.filter(user_profile=user, semester=semester).aggregate(Sum('value'))['value__sum'] or 0
```

Comment:
Why always create a new object and not just update the old one?

Reply:
As rewards are tied to financial operation at events, these objects shall not be modified. Ask @janno42 for a more detailed explanation.

Reply:
Alrighty, that's a point